src/main/java/com/yugabyte/sample/apps/SqlBankTransfers.java

```diff
@@ -0,0 +1,229 @@
+ * 2. Start a repeatable read transaction.
+ * 3. Query the account balance of the sender.
+ * 4. If the balance is zero, abort the transaction.
+ * 5. Pick a random amount [1, balance].
+ * 6. Decrement the balance of the sender by the amount.
+ * 7. Increment the balance of the receiver by the amount.
+ * 8. Commit the transaction.
+ *
+ * Verify Operation:
+ * 1. Sum the balances of all accounts.
+ * 2. Verify that the sum is 1000 * 1000.
+ */
+public class SqlBankTransfers extends AppBase {
+  private static final Logger LOG = Logger.getLogger(SqlBankTransfers.class);
+
+  // Static initialization of this app's config.
+  static {
+    // Use 1 Verify thread and 10 Transfer threads.
+    appConfig.readIOPSPercentage = -1;
+    appConfig.numReaderThreads = 1;
+    appConfig.numWriterThreads = 10;
+    // Disable number of keys.
+    appConfig.numKeysToRead = -1;
+    appConfig.numKeysToWrite = -1;
+    // Run the app for 1 minute.
+    appConfig.runTimeSeconds = 60;
+    // Report restart read requests metric by default.
+    appConfig.restartReadsReported = true;
+    // Avoid load balancing errors.
+    appConfig.loadBalance = false;
+    appConfig.disableYBLoadBalancingPolicy = true;
+  }
+
+  // The default table name to create and use for ops.
+  private static final String DEFAULT_TABLE_NAME = "bank_accounts";
+
+  // The number of accounts in the bank.
+  private static final int NUM_ACCOUNTS = 1000;
+
+  // Initial balance of each account.
+  private static final int INIT_BALANCE = 1000;
+
+  // Shared counter to store the number of inconsistent reads.
+  private static AtomicLong numInconsistentReads = new AtomicLong(0);
+
+  @Override
+  public void createTablesIfNeeded(TableOp tableOp) throws Exception {
+    try (Connection connection = getPostgresConnection()) {
+      // Every run should start cleanly.
+      connection.createStatement().execute(
+          String.format("DROP TABLE IF EXISTS %s", getTableName()));
+      LOG.info("Dropping any table(s) left from previous runs if any");
+      connection.createStatement().execute(String.format(
+          "CREATE TABLE %s (account_id INT, balance INT)",
+          getTableName()));
+      LOG.info(String.format("Created table: %s", getTableName()));
+      int numRows = connection.createStatement().executeUpdate(String.format(
+          "INSERT INTO %s SELECT GENERATE_SERIES(0, %d-1), %d",
+          getTableName(), NUM_ACCOUNTS, INIT_BALANCE));
+      LOG.info(String.format(
+          "Inserted %d rows into %s", numRows, getTableName()));
+    }
+  }
+
+  @Override
+  public String getTableName() {
+    String tableName = appConfig.tableName != null ?
+        appConfig.tableName : DEFAULT_TABLE_NAME;
+    return tableName.toLowerCase();
+  }
+
+  // Executes the Verify operation.
+  @Override
+  public long doRead() {
+    try (Connection connection = getPostgresConnection();
+         Statement statement = connection.createStatement()) {
```

Comment:
Note that every read and write in this case will create new connection - if we want performance here, we should hash connections. Otherwise it's ~300ms additionally to each operation just to establish connection.

Reply:
Or prepareStatement as in other workloads

Reply:
Thanks for the suggestion. I pick connections from a pool now.

I cannot use prepared statements like in other workloads because I need concurrency. Other workloads are using just one prepared statement and that serializes the operations.

Reply:
Spoke with qvad offline. I had the wrong mental model that there is only a single app across threads. Let me update them to use prepared statements.

Reply:
Updated with prepared statements wherever possible.